dds/DCPS/ThreadMonitor.h

```diff
@@ -0,0 +1,81 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_THREAD_MONITOR_H
+#define OPENDDS_DCPS_THREAD_MONITOR_H
+
+#include <string>
+
+namespace OpenDDS {
+namespace DCPS {
```

Comment:
Support for versioned namespaces needed here (and in matching .cpp), see other files for examples